### PR TITLE
wip: test for chat request payment stellar messages

### PR DESCRIPTION
**Pull request by zapu**
- [x] do the revendor dane and rebase

### CODE REVIEW

File: go/chat/server_test.go [hunk 1]
```diff
@@ -5364,6 +5365,67 @@ func TestChatSrvGetStaticConfig(t *testing.T) {
...
 
+func TestChatSrvStellarMessages(t *testing.T) {
+	runWithMemberTypes(t, func(mt chat1.ConversationMembersType) {
+		runWithEphemeral(t, mt, func(ephemeralLifetime *gregor1.DurationSec) {
+			switch mt {
+			case chat1.ConversationMembersType_KBFS:
+				return
+			}
+
+			ctc := makeChatTestContext(t, "PostLocalNonblock", 2)
```
Comment (joshblum) [inline go/chat/server_test.go]: `SrvStellarMessages`

File: go/chat/server_test.go [hunk 2]
```diff
@@ -5364,6 +5365,67 @@ func TestChatSrvGetStaticConfig(t *testing.T) {
...
+			defer ctc.cleanup()
+			users := ctc.users()
+
+			tc := ctc.as(t, users[0])
+			listener := newServerChatListener()
+			ctc.as(t, users[0]).h.G().NotifyRouter.SetListener(listener)
+			ctc.world.Tcs[users[0].Username].ChatG.Syncer.(*Syncer).isConnected = true
+
+			created := mustCreateConversationForTest(t, ctc, users[0], chat1.TopicType_CHAT,
+				chat1.ConversationMembersType_IMPTEAMNATIVE, ctc.as(t, users[1]).user())
```
Comment (joshblum) [inline go/chat/server_test.go]: this should use `mt` or we're not really testing all of the member types

File: go/chat/server_test.go [hunk 3]
```diff
@@ -5364,6 +5365,67 @@ func TestChatSrvGetStaticConfig(t *testing.T) {
...
+			darg := chat1.PostDeleteNonblockArg{
+				ConversationID:   created.Id,
+				TlfName:          created.TlfName,
+				TlfPublic:        created.Visibility == keybase1.TLFVisibility_PUBLIC,
+				Supersedes:       unboxed.GetMessageID(),
+				IdentifyBehavior: keybase1.TLFIdentifyBehavior_CHAT_CLI,
+			}
+			res, err := ctc.as(t, users[0]).chatLocalHandler().PostDeleteNonblock(tc.startCtx, darg)
+			require.NoError(t, err)
+			_ = res
```
Comment (joshblum) [inline go/chat/server_test.go]: load the thread and verify the message is deleted

File: go/chat/server_test.go [hunk 4]
```diff
@@ -5364,6 +5365,77 @@ func TestChatSrvGetStaticConfig(t *testing.T) {
...
+
+			_, err := postLocalForTestNoAdvanceClock(t, ctc, users[0], created, body)
+			require.NoError(t, err)
+
+			var unboxed chat1.UIMessage
+			select {
+			case info := <-listener.newMessageRemote:
+				unboxed = info.Message
+				require.True(t, unboxed.IsValid(), "invalid message")
+				require.Equal(t, chat1.MessageType_REQUESTPAYMENT, unboxed.GetMessageType(), "invalid type")
```
Comment (joshblum) [inline go/chat/server_test.go]: can you verify the contents are what you posted here: https://github.com/keybase/client/pull/13421/files#diff-af0e80292e0633ef269b31efe7e7ffcfR5389

Comment (zapu) [reply go/chat/server_test.go]: done below

**Pull request merged**